java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java

```diff
@@ -194,7 +194,8 @@ public void takeState(String clientFeatures) throws YggdrasilInvalidInputExcepti
       byte[] messageBytes = clientFeatures.getBytes(StandardCharsets.UTF_8);
       nativeInterface.takeState(this.enginePointer, messageBytes);
     } catch (RuntimeException e) {
-      throw new YggdrasilInvalidInputException("Failed to take state:", e);
+      throw new YggdrasilInvalidInputException(
```

Comment:
I don't think this is correct here. The passed data being an invalid state file won't trigger this exception, this only occurs if something goes wrong with the internals of Chicory